server/src/Server.cpp

```diff
@@ -446,30 +472,40 @@ void Server::handleRequest(int clientFd) // i is the index from the evList of th
 			//determine location
 			request.setDecodedTarget(percentDecoding(request.getRawTarget()));
 			request.setQuery(percentDecoding(request.getQuery()));
-			isCgi = this->_isCgiRequest(this->_requestHead);
+			this->_response.setIsCgi(clientFd, this->_isCgiRequest(this->_requestHead));
 			#ifdef SHOW_LOG
 				std::cout  << YELLOW << "URI after percent-decoding: " << request.getDecodedTarget() << std::endl;
 			#endif
 			this->matchLocation(request); // AE location with ü (first decode only unreserved chars?)
 			// request.setRoutedTarget("." + request.getRoutedTarget());
 			//check method
-			checkLocationMethod(request);
-
-			if (request.getHeaderFields().count("connection") && request.getHeaderFields().find("connection")->second == "keep-alive")
+			if (isCgi == false)
+				this->checkLocationMethod(request);
+			if ((request.getHeaderFields().count("connection") && request.getHeaderFields().find("connection")->second == "keep-alive") || request.getHeaderFields().count("connection") == 0)
 				this->_socketHandler->addKeepAlive(clientFd);
 
-			if (isCgi == true)
+			/* if (isCgi == true)
 			{
 				this->applyCurrentConfig(request);
 				cgi_handle(request, clientFd, this->_currentConfig);
 			}
-			else if (request.getMethod() == "POST" || request.getMethod() == "PUT")
+			else  */if (request.getMethod() == "POST" || request.getMethod() == "PUT")
```

Comment:
check if only enabled for tester

Reply:
9ecbfbcd4f8b6eca5c98ebb9a119f2d6a6b4ae08 fixed it